memory/transparent_hugepages.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Test Inspired basically from
+# https://github.com/autotest/tp-qemu/blob/master/generic/tests/trans_hugepage.py
+#
+# Copyright: 2017 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.core import data_dir
+from avocado.utils.partition import Partition
+
+
+class Thp(Test):
+
+    '''
+    The test enables THP and stress the system using dd load
+    and verifies whether THP has been allocated for usage or not
+    '''
+
+    def setUp(self):
+
+        '''
+        Sets all the reqd parameter and also
+        mounts the device to be used in test.
+        '''
+
+        # Set params as per available memory in system
+        self.tmpdir = data_dir.get_tmp_dir()
```

Comment:
No need to create it as an instance variable as `self.tmpdir` as is not used outside of `setUp` routine any i think you can drop it it can be like 

`self.mem_path = os.path.join(data_dir.get_tmp_dir(), 'thp_space')`